spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/config/AwsClientProperties.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.aws.core.config;
+
+import java.net.URI;
+
+/**
+ * Base properties for AWS Service client.
+ *
+ * @author Maciej Walkowiak
+ */
+public abstract class AwsClientProperties {
```

Comment:
I like this but I think we could create a PR just introducing this change and later a new PR introducing the endpoint property.